cccatalog-api/cccatalog/api/views/license_views.py

```diff
@@ -0,0 +1,25 @@
+from django.core import serializers
+from django.http import HttpResponse
+from django.views import View
+from django.views.generic import ListView
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import viewsets
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from cccatalog.api.models import Image
+from cccatalog.api.serializers.license_serializers import LicenseSerializer
+
+class ListLicense(APIView):
+    """
+    List all licenses available in catalog
+    """
+    @swagger_auto_schema(operation_id='license',
+                         query_serializer=LicenseSerializer,
+                         responses={
+                             200: LicenseSerializer(many=True)
+                         })
+    def get(self, request):
+        images = Image.objects.order_by('license', 'license_version').distinct('license', 'license_version')
```

Comment:
You have the right idea here, but the problem is that performing `distinct` on 330 million+ records will be very slow (in the realm of hours or days). I suggest taking a look at `licenses.py`, which is more or less our hardcoded internal model for available licenses.

I also suggest exposing the license URL.

Reply:
Thanks @aldenstpage for quick review. I did not have information about database size. As you rightly pointed out that `distinct` on huge number of records will be an expensive operation. I had a look at `licenses.py`. I will use that to create response.

I did not get your point about *exposing the license URL*

Reply:
In `licenses.py`, there's a string called `LICENSE_URL` that can be formatted to be a direct link to the licenses - it would be useful to return this from the `/licenses` endpoint as well

Reply:
`LICENSES` tuple in `licenses.py` holds list of licenses. But we don't have version numbers. Where to get that from?
Response should be in form of 
```
[{
'license': 'License name',
'license_version': 2,
url: 'license_url'
}, 
...]